src/parsing.jl

```diff
@@ -214,9 +214,9 @@ _DEFAULT_TYPE_MAP[:timestamp] = DateTime
 const TIMESTAMP_FORMAT = dateformat"y-m-d HH:MM:SS.s"  # .s is optional here
 function pqparse(::Type{DateTime}, str::AbstractString)
     if str == "infinity"
-        return InfExtendedTime{DateTime}(∞)
+        return typemax(DateTime)
```

Comment:
After consideration for existing users I don't think we should remove this entirely but I do think we should deprecate it. In the infinity cases, add some kind of ad hoc warn-once function using a global `Ref{Bool}` to give a deprecation warning to users that directs them to use `InfExtendedTime` if they want `infinity` support for later major versions of LibPQ.jl.